modules/rostests/apitests/partmgr/StorDeviceNumber.c

```diff
@@ -0,0 +1,304 @@
+/*
+ * PROJECT:     ReactOS API Tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Test for IOCTL_STORAGE_GET_DEVICE_NUMBER
+ * COPYRIGHT:   Copyright 2024 Hermès Bélusca-Maïto <hermes.belusca-maito@reactos.org>
+ */
+
+#include "precomp.h"
+#include <ntddstor.h>
+
+/* Flags combination allowing all the read, write and delete share modes.
+ * Currently similar to FILE_SHARE_VALID_FLAGS. */
+#define FILE_SHARE_ALL \
+    (FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE)
+
+static BOOLEAN
+Test_Device_StorDeviceNumber(
+    _In_ PCWSTR NtDeviceName)
+{
+    BOOLEAN Success = FALSE; // Suppose failure.
+    NTSTATUS Status;
+    HANDLE DeviceHandle1 = NULL, DeviceHandle2 = NULL;
+    FILE_FS_DEVICE_INFORMATION DeviceInfo;
+    STORAGE_DEVICE_NUMBER DeviceNumber;
+    UNICODE_STRING DeviceName;
+    OBJECT_ATTRIBUTES ObjectAttributes;
+    IO_STATUS_BLOCK IoStatusBlock;
+    WCHAR NtLegacyDeviceName[MAX_PATH];
+
+    ULONG BufferSize;
+    struct { OBJECT_NAME_INFORMATION; WCHAR Buffer[MAX_PATH]; } DeviceName1Buffer;
+    PUNICODE_STRING DeviceName1 = &DeviceName1Buffer.Name;
+    struct { OBJECT_NAME_INFORMATION; WCHAR Buffer[MAX_PATH]; } DeviceName2Buffer;
+    PUNICODE_STRING DeviceName2 = &DeviceName2Buffer.Name;
+
+    /* Open a handle to the device */
+    RtlInitUnicodeString(&DeviceName, NtDeviceName);
+    InitializeObjectAttributes(&ObjectAttributes,
+                               &DeviceName,
+                               OBJ_CASE_INSENSITIVE,
+                               NULL,
+                               NULL);
+    Status = NtOpenFile(&DeviceHandle1,
+                        FILE_READ_ATTRIBUTES | SYNCHRONIZE,
+                        &ObjectAttributes,
+                        &IoStatusBlock,
+                        FILE_SHARE_ALL,
+                        /* FILE_NON_DIRECTORY_FILE | */ FILE_SYNCHRONOUS_IO_NONALERT);
+    ok_ntstatus(Status, STATUS_SUCCESS);
+    if (!NT_SUCCESS(Status))
+    {
+        skip("Device '%wZ': Opening failed\n", &DeviceName);
```

Comment:
`error: unknown conversion type character 'w' in format [-Werror=format=]`

Reply:
Weird, why does it complain here? Printf8ng unicode string structures is normally supported...